### DISCUSSION TITLE
Update Onboarding config for open merge request

### DISCUSSION THREAD

**Discussion by MPritsch** [renovatebot/renovate#25119] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.6.1

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I'm new to renovate. I'm in the process of creating an onboarding configuration, specifically with credentials to a private maven registry. I initially ran the default renovate gitlab-ci template which onboarded my repository with the default config. Since I want to change that config, I changed every place that looks like it could change the config in the merge-request. So far nothing seems to change this config. The lookups of our maven dependency fails, since it tries to access the public maven registry. The gitlab is in a private subnet, so it can't actually reach public registries.

I expected the endresult to change the renovate config of an open Onboarding merge-request in a repository. This should happen when I update the onboardingConfig in my config.js or the "RENOVATE_ONBOARDING_CONFIG" variable of my renovate-runner repository and rerun the renovate job.

Since I'm new, I don't know if this would work, or how I can update it anyway. Can you delete the "RENOVATE_ONBOARDING_CONFIG" variable and the config will be pulled from the config.js?

Is the renovate onboarding config of an open onboarding merge request ever updated anyway? Or is the only way to manually edit that MR? I wanted this repository to be the pilot of the group, so I wanted to see that it works there first.

### What you tried so far.

* Use different config in the RENOVATE_ONBOARDING_CONFIG variable
* Use different config in the "onboardingConfig" of the config.js in the renovate-runner repository
* Remove the "RENOVATE_ONBOARDING_CONFIG" variable
* Disable "RENOVATE_REPOSITORY_CACHE" variable

### Relevant debug logs

In the example I changed the "RENOVATE_ONBOARDING_CONFIG" to use a local config

<details><summary>Logs</summary>

```
RENOVATE_ONBOARDING_CONFIG: '{"$$schema": "https://docs.renovatebot.com/renovate-schema.json", "extends": ["config:recommended","local>path/to/renovate-runner:not-onboarded"] }'
```

In the MR this config is like this:
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["config:recommended"]
}
```

The logs of the renovate job:
```
$ renovate $RENOVATE_EXTRA_FLAGS
 WARN: Config needs migrating
       "originalConfig": {
         "onboarding": "true",
         "onboardingConfig": {
           ... custom onboarding config
         },
         "platform": "gitlab"
       },
       "migratedConfig": {
         "onboarding": true,
         "onboardingConfig": {
           ... custom onboarding config
         },
         "platform": "gitlab"
       }


 INFO: Autodiscovered repositories
       "length": 1,
       "repositories": ["path/to/onboarding/repository"]
 INFO: Repository started (repository=path/to/onboarding/repository)
       "renovateVersion": "37.6.1"
```

</details>

**Comment by Sax388** [comment]
I would be interested in an answer (to the question, how to update an Onboarding MR after it has been created) here. I found this https://github.com/renovatebot/config-help/issues/54 here (renaming the Onboarding MR to get a new one) but I haven't tested it yet.

**Reply by Sax388** [reply]
Now I did test it and it didn't work. Then I found https://github.com/renovatebot/renovate/discussions/31874 where I understood (from https://github.com/renovatebot/renovate/discussions/31874#discussioncomment-10897060) that you have not only to rename the MR but to close it as well. Let's see how that goes.

**Reply by Sax388** [reply]
Deleting the branch together with the rename did the trick. I still don't understand why renovate doesn't just update the onboarding MRs like the others.
